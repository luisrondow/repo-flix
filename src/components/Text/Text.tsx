type TextProps = {
  as?: 'h1' | 'h2' | 'p' | 'span'
  children: React.ReactNode
  className?: string
}

const Text = (props: TextProps) => {
  const { children, as, className: customClassName } = props

  const className = `${customClassName ?? ''} text-[#e0def4] font-roboto`

  switch (as) {
    case 'h1':
      return <h1 className={`${className} text-4xl font-bold`}>{children}</h1>
    case 'h2':
      return <h2 className={`${className} text-xl font-bold`}>{children}</h2>
    case 'p':
      return <p className={className}>{children}</p>
    case 'span':
      return <span className={className}>{children}</span>
    default:
      return <p className={className}>{children}</p>
  }
}

export default Text
