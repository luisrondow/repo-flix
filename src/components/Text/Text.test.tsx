import { render } from '@testing-library/react'
import Text from './Text'

describe('Text', () => {
  const tags = ['h1', 'h2', 'p', 'span']

  it('should render correctly', () => {
    const { asFragment, getByText } = render(<Text>Hello World</Text>)

    expect(getByText('Hello World')).toBeInTheDocument()
    expect(asFragment()).toMatchSnapshot()
  })

  tags.forEach((tag) => {
    it(`should render with tag ${tag}`, () => {
      const { asFragment, getByText, container } = render(
        <Text as={tag as 'h1' | 'h2' | 'p' | 'span'}>Hello World</Text>,
      )

      expect(getByText('Hello World')).toBeInTheDocument()
      expect(container.querySelector(tag)).toBeInTheDocument()
      expect(asFragment()).toMatchSnapshot()
    })
  })
})
