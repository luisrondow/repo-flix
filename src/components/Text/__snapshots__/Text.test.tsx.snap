// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text should render correctly 1`] = `
<DocumentFragment>
  <p
    class=" text-[#e0def4] font-roboto"
  >
    Hello World
  </p>
</DocumentFragment>
`;

exports[`Text should render with tag h1 1`] = `
<DocumentFragment>
  <h1
    class=" text-[#e0def4] font-roboto text-4xl font-bold"
  >
    Hello World
  </h1>
</DocumentFragment>
`;

exports[`Text should render with tag h2 1`] = `
<DocumentFragment>
  <h2
    class=" text-[#e0def4] font-roboto text-xl font-bold"
  >
    Hello World
  </h2>
</DocumentFragment>
`;

exports[`Text should render with tag p 1`] = `
<DocumentFragment>
  <p
    class=" text-[#e0def4] font-roboto"
  >
    Hello World
  </p>
</DocumentFragment>
`;

exports[`Text should render with tag span 1`] = `
<DocumentFragment>
  <span
    class=" text-[#e0def4] font-roboto"
  >
    Hello World
  </span>
</DocumentFragment>
`;
